manifests/init.pp

```diff
@@ -229,53 +254,53 @@
-  $package_corosync                    = $::corosync::params::package_corosync,
-  $package_crmsh                       = $::corosync::params::package_crmsh,
-  $package_pacemaker                   = $::corosync::params::package_pacemaker,
-  $package_pcs                         = $::corosync::params::package_pcs,
-  $packageopts_corosync                = $::corosync::params::package_install_options,
-  $packageopts_pacemaker               = $::corosync::params::package_install_options,
-  $packageopts_crmsh                   = $::corosync::params::package_install_options,
-  $packageopts_pcs                     = $::corosync::params::package_install_options,
-  $version_corosync                    = $::corosync::params::version_corosync,
-  $version_crmsh                       = $::corosync::params::version_crmsh,
-  $version_pacemaker                   = $::corosync::params::version_pacemaker,
-  $version_pcs                         = $::corosync::params::version_pcs,
-  $set_votequorum                      = $::corosync::params::set_votequorum,
-  $votequorum_expected_votes           = $::corosync::params::votequorum_expected_votes,
-  $quorum_members                      = ['localhost'],
-  $quorum_members_ids                  = undef,
-  $token                               = $::corosync::params::token,
-  $token_retransmits_before_loss_const = $::corosync::params::token_retransmits_before_loss_const,
-  $compatibility                       = $::corosync::params::compatibility,
-  $manage_pacemaker_service            = $::corosync::params::manage_pacemaker_service,
-  $manage_pcsd_service                 = false,
-  $cluster_name                        = $::corosync::params::cluster_name,
-  $join                                = $::corosync::params::join,
-  $consensus                           = $::corosync::params::consensus,
-  $max_messages                        = $::corosync::params::max_messages,
-  $test_corosync_config                = $::corosync::params::test_corosync_config,
+  Enum['on', 'off'] $enable_secauth                                           = $::corosync::params::enable_secauth,
+  Enum['file', 'string'] $authkey_source                                      = $::corosync::params::authkey_source,
+  String $authkey                                                             = $::corosync::params::authkey,
+  Optional[Integer] $threads                                                  = $::corosync::params::threads,
+  String $port                                                                = $::corosync::params::port,
+  Variant[String, Array] $bind_address                                        = $::corosync::params::bind_address,
+  Optional[String] $multicast_address                                         = $::corosync::params::multicast_address,
+  Optional[Array] $unicast_addresses                                          = $::corosync::params::unicast_addresses,
```

Comment:
for IP addresses, you can use `Stdlib::Compat::Ip_address`. This will work even with the variant, though it will be long, e.g. (untested):
`Variant[String[Stdlib::Compat::Ip_address], Array[Stdlib::Compat::Ip_address]`